test/gateway/index.js

```diff
@@ -0,0 +1,213 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+const series = require('async/series')
+const API = require('../../src/http')
+const loadFixture = require('aegir/fixtures')
+const os = require('os')
+
+const bigFile = loadFixture(__dirname, '../../node_modules/interface-ipfs-core/test/fixtures/15mb.random', 'ipfs')
+const directoryContent = {
+  'index.html': loadFixture(__dirname, './test-folder/index.html', 'ipfs'),
+  'nested-folder/hello.txt': loadFixture(__dirname, './test-folder/nested-folder/hello.txt', 'ipfs'),
+  'nested-folder/ipfs.txt': loadFixture(__dirname, './test-folder/nested-folder/ipfs.txt', 'ipfs'),
+  'nested-folder/nested.html': loadFixture(__dirname, './test-folder/nested-folder/nested.html', 'ipfs')
+}
+
+describe('HTTP Gateway', () => {
+  let http = {}
+  let gateway
+
+  before((done) => {
+    const repoPath = os.tmpdir() + '/ipfs-' + Math.random().toString().substring(2, 8) + '-' + Date.now()
+
+    http.api = new API(repoPath, {
```

Comment:
If you use the default-config found in either ipfs-factory or ipfs-instance (directories inside the test-directory), you'll get a perfectly fine config to use.